### PR TITLE
[CSM] infer codec model with no_grad + audio eos label

### PR DESCRIPTION
**Pull request by eustlb**
The codec model should be inferred with torch.no_grad()!
While this has no effect on training results, since the backpropagated gradients ultimately go to zero (as the loss is computed from the output of the codec model), the codec model’s parameters still remain part of the computational, gradients are unnecessarily computed.

It also makes ensure the audio eos is not masked with -100 when preparing the target label with the processor.

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Indeed, in a way we should probably disable gradient by default on these models!

**Comment by eustlb** [comment]
That's amazing! Happy to amplify comms on it when you publish 😉
